src/PointwiseFunctions/AnalyticSolutions/Xcts/ConstantDensityStar.hpp

```diff
@@ -0,0 +1,189 @@
+    static constexpr OptionString help{"The constant density within the star"};
+    static double lower_bound() noexcept { return 0.; }
+  };
+  struct Radius {
+    using type = double;
+    static constexpr OptionString help{"The conformal radius of the star"};
+    static double lower_bound() noexcept { return 0.; }
+  };
+
+ public:
+  using options = tmpl::list<Density, Radius>;
+  static constexpr OptionString help{
+      "A constant density star in general relativity"};
+
+  ConstantDensityStar() = default;
+  ConstantDensityStar(const ConstantDensityStar&) noexcept = delete;
+  ConstantDensityStar& operator=(const ConstantDensityStar&) noexcept = delete;
+  ConstantDensityStar(ConstantDensityStar&&) noexcept = default;
+  ConstantDensityStar& operator=(ConstantDensityStar&&) noexcept = default;
+  ~ConstantDensityStar() noexcept = default;
+
+  ConstantDensityStar(double density, double radius,
+                      const OptionContext& context = {});
+
+  double density() const noexcept { return density_; }
+  double radius() const noexcept { return radius_; }
+
+  // @{
+  /// Retrieve variable at coordinates `x`
+  auto variables(
+      const tnsr::I<DataVector, 3, Frame::Inertial>& x,
+      tmpl::list<Xcts::Tags::ConformalFactor<DataVector>> /*meta*/) const
+      noexcept -> tuples::TaggedTuple<Xcts::Tags::ConformalFactor<DataVector>>;
+
+  static auto variables(
+      const tnsr::I<DataVector, 3, Frame::Inertial>& x,
+      tmpl::list<::Tags::Initial<
+          Xcts::Tags::ConformalFactor<DataVector>>> /*meta*/) noexcept
+      -> tuples::TaggedTuple<
+          ::Tags::Initial<Xcts::Tags::ConformalFactor<DataVector>>>;
+
+  static auto variables(
+      const tnsr::I<DataVector, 3, Frame::Inertial>& x,
+      tmpl::list<::Tags::Initial<Xcts::Tags::ConformalFactorGradient<
+          3, Frame::Inertial, DataVector>>> /*meta*/) noexcept
+      -> tuples::TaggedTuple<::Tags::Initial<
+          Xcts::Tags::ConformalFactorGradient<3, Frame::Inertial, DataVector>>>;
+
+  static auto variables(
+      const tnsr::I<DataVector, 3, Frame::Inertial>& x,
+      tmpl::list<::Tags::Source<
+          Xcts::Tags::ConformalFactor<DataVector>>> /*meta*/) noexcept
+      -> tuples::TaggedTuple<
+          ::Tags::Source<Xcts::Tags::ConformalFactor<DataVector>>>;
+
+  static auto variables(
+      const tnsr::I<DataVector, 3, Frame::Inertial>& x,
+      tmpl::list<::Tags::Source<Xcts::Tags::ConformalFactorGradient<
+          3, Frame::Inertial, DataVector>>> /*meta*/) noexcept
+      -> tuples::TaggedTuple<::Tags::Source<
+          Xcts::Tags::ConformalFactorGradient<3, Frame::Inertial, DataVector>>>;
+
+  auto variables(const tnsr::I<DataVector, 3, Frame::Inertial>& x,
+                 tmpl::list<gr::Tags::EnergyDensity<DataVector>> /*meta*/) const
+      noexcept -> tuples::TaggedTuple<gr::Tags::EnergyDensity<DataVector>>;
+  // @}
+
+  /// Retrieve a collection of variables at coordinates `x`
+  template <typename... Tags>
```

Comment:
@nilsdeppe please discuss with @nilsleiffischer if this interface should be consistent with the ones for evolution systems, and if so which interface is the one to be used...

Reply:
We were not sure what you meant was inconsistent between the interfaces. The only difference is that the elliptic analytic solutions don't have a `time` argument. I added template specialisations for `double` and `DataVector` now because we figured why not. It's in the fixup commit so could you look at that please before I squash?

Reply:
In some analytic solutions, there are two public `variables` one taking a typelist of Tags, and another taking a single Tag, both forwarding to private `variables` that act on a specific Tag.  In others there is something similar to here where there is one public `variables` taking a typelist of Tags with a static assert about infinite recursion and public specific single Tags `variables` . It is not clear to me why we have these two different styles

Reply:
In the case where there is an intermediate computation that is expensive and done for all variables we have the 2 public variables version where one takes one tag the other multiple. In the case where there is no expensive intermediate computation we can just have the variables for each member be public.

Reply:
I think these two implementation styles result in the same public interface, correct? Looks to me that the generic `variables` that takes a single tag and forwards to a private function is only needed to compute `IntermediateVariables`.

Reply:
@nilsdeppe was faster

Reply:
okay, thanks